proc-macros/src/render_server.rs

```diff
@@ -200,20 +222,39 @@ impl RpcDescription {
 				};
 
 				if sub.signature.sig.asyncness.is_some() {
-					handle_register_result(quote! {
-						rpc.register_subscription(#rpc_sub_name, #rpc_notif_name, #rpc_unsub_name, |params, mut pending, context, ext| async move {
-							#parsing
-							#into_sub_response::into_response(context.as_ref().#rust_method_name(pending, &ext, #params_seq).await)
+					if sub.with_extensions {
+						handle_register_result(quote! {
+							rpc.register_subscription(#rpc_sub_name, #rpc_notif_name, #rpc_unsub_name, |params, mut pending, context, ext| async move {
+								#parsing
+								#into_sub_response::into_response(context.as_ref().#rust_method_name(pending, &ext, #params_seq).await)
+							})
 						})
-					})
+					} else {
+						handle_register_result(quote! {
+							rpc.register_subscription(#rpc_sub_name, #rpc_notif_name, #rpc_unsub_name, |params, mut pending, context, _| async move {
+								#parsing
+								#into_sub_response::into_response(context.as_ref().#rust_method_name(pending, #params_seq).await)
+							})
+						})
+					}
 				} else {
-					handle_register_result(quote! {
-						rpc.register_subscription_raw(#rpc_sub_name, #rpc_notif_name, #rpc_unsub_name, |params, mut pending, context, ext| {
-							#parsing
-							let _ = context.as_ref().#rust_method_name(pending, &ext, #params_seq);
-							#sub_err::None
+					if sub.with_extensions {
```

Comment:
can you collapse the `if` into `else if` following the recommendation from clippy, otherwise LGTM